### DISCUSSION TITLE
Is source of the documentation available in any human-readable format

### DISCUSSION THREAD

**Discussion by SomeoneElseOSM** [maplibre/maplibre-gl-js#5256] [General]
Currently, the presentation of the data on https://maplibre.org/maplibre-gl-js/docs/API/ isn't very useful.  There are three columns of data, each with their own scroll bars, and only one of which is useful at any one time depending on the information being searched for.  The search box at the top of the screen is at best unhelpful, so in-page search is the only option - which of course doesn't work unless you can get rid of the other two columns.

https://github.com/maplibre/maplibre-gl-js/discussions/2148 suggests that the source for the current pages are held elsewhere .  Where is that and what other options are there for presenting it?  Even just a series of interlinked text files would be better than https://maplibre.org/maplibre-gl-js/docs/API/ and the even less helpful https://maplibre.org/maplibre-gl-js/docs/examples/ .

**Comment by HarelM** [comment]
Since the linked discussion was done the documentation have been completely rewritten using typedoc and kmdocs material and are generated as part of the build process in this repo. 
I find it very useful, informative and easy to maintain. 
I understand that you don't, but the comment above isn't helpful at all at actually making the docs better, but instead, poops on the hard work of the maintainers here. 
If you have a good idea how to improve it feel free to suggest it, saying something is bad isn't helping anyone...

**Reply by SomeoneElseOSM** [reply]
I'm not criticizing the documentation itself (I can't do that, since I've not found it yet, other than view presented on the web pages linked above).  Where actually is it?

**Reply by HarelM** [reply]
It is generated mostly. You can read on it here:
https://github.com/maplibre/maplibre-gl-js/tree/main/docs#readme

**Reply by SomeoneElseOSM** [reply]
So apart from the short pages at https://github.com/maplibre/maplibre-gl-js/tree/main/test/examples there isn't any human-written documentation?  That explains a lot of what I am seeing.

However, presumably the thing that collates the tsdoc comments is separate from the thing that determines that layout of https://maplibre.org/maplibre-gl-js/docs/API/ - it's the layout of that that basically makes it unnavigable and unsearchable via ^f.  Presumably there's a mkdocs layout somewhere, and presumably I could change that (locally) to only produce one column of data?  That would at least be searchable.

The question that I was trying to answer when I started start this discussion was "how do I find the latitude and longitude of the current view of a Maplibre GL map"  To give an example of the problem with the search box at https://maplibre.org/maplibre-gl-js/docs/API , try searching for that and you get https://maplibre.org/maplibre-gl-js/docs/examples/add-image-animated/ which is at best unrelated.  Search again for "latitude" and you get the same page again - because it's not searching the API documentation, but some other part of the site.  If you _manually_ navigate back to https://maplibre.org/maplibre-gl-js/docs/API/ and try searching for something much simpler like "latitude" the actual answer is there, but only as the 12th answer of 14, because the actual text associated with the "getCenter()" method does not contain that word.  Searching for "center" doesn't work because that word is used almost everywhere in different contexts (geolocation etc.) - there are 116 answers to that,

So I guess here are the problems I'm seeing:

- There aren't enough actual words in the actual documentation (presumably these are from tsdoc comments).
- The search box at https://maplibre.org/maplibre-gl-js/docs/API/ doesn't just search the API, and there is no effective way of just searching the API
- The layout of https://maplibre.org/maplibre-gl-js/docs/API/ (I presume a mkdocs layout) makes searching the data there or even just reading it difficult.

**Reply by HarelM** [reply]
The documentation is SEO friendly, did you try searching things using google?
The search can be improved, no doubt, but searching a mapping library for "latitude", which is a very generic term, and expecting valid results is a bit ambitious, IMHO.
Having said that, you are welcome to try and invest time in improving the docs, I'm probably too used to it to have a good observation of what's not working as expected and what can be improved...

**Reply by neodescis** [reply]
> So apart from the short pages at https://github.com/maplibre/maplibre-gl-js/tree/main/test/examples there isn't any human-written documentation? That explains a lot of what I am seeing.
> 
> However, presumably the thing that collates the tsdoc comments is separate from the thing that determines that layout of https://maplibre.org/maplibre-gl-js/docs/API/ - it's the layout of that that basically makes it unnavigable and unsearchable via ^f. Presumably there's a mkdocs layout somewhere, and presumably I could change that (locally) to only produce one column of data? That would at least be searchable.
> 
> The question that I was trying to answer when I started start this discussion was "how do I find the latitude and longitude of the current view of a Maplibre GL map" To give an example of the problem with the search box at https://maplibre.org/maplibre-gl-js/docs/API , try searching for that and you get https://maplibre.org/maplibre-gl-js/docs/examples/add-image-animated/ which is at best unrelated. Search again for "latitude" and you get the same page again - because it's not searching the API documentation, but some other part of the site. If you _manually_ navigate back to https://maplibre.org/maplibre-gl-js/docs/API/ and try searching for something much simpler like "latitude" the actual answer is there, but only as the 12th answer of 14, because the actual text associated with the "getCenter()" method does not contain that word. Searching for "center" doesn't work because that word is used almost everywhere in different contexts (geolocation etc.) - there are 116 answers to that,
> 
> So I guess here are the problems I'm seeing:
> 
> * There aren't enough actual words in the actual documentation (presumably these are from tsdoc comments).
> * The search box at https://maplibre.org/maplibre-gl-js/docs/API/ doesn't just search the API, and there is no effective way of just searching the API
> * The layout of https://maplibre.org/maplibre-gl-js/docs/API/ (I presume a mkdocs layout) makes searching the data there or even just reading it difficult.

Your expectations here are, quite simply, ludicrous. If you're looking for an LLM integration into the docs, perhaps you should come up with a plan to add it.

I do generally agree that finding things in the documentation can be difficult for a newcomer, but offering suggestions rather than bellyaching will go a long way.

**Reply by SomeoneElseOSM** [reply]
I'm not looking for an LLM integration into the docs. I really don't see how anything that I've written above could be interpreted that way.

> suggestions rather than bellyaching 

Ahem.  I asked at the top of this discussion where the source of the documentation is. I now know that it's in a bunch of tsdoc comments somewhere, but I still don't know where that "somewhere" is. No-one can help improve anything unless the thing that needs improvement can actually be identified.

I appreciate that you're feeling unhappy because someone has had the nerve to criticise something that you have volunteered your time on. However, that sort of reaction really isn't helpful. Maybe there isn't a problem (maybe this documentation isn't supposed to be used, other than in a "I know what all the methods are but what are the parameters) sense, but if it is, there _is_ a problem.

There are lots of projects out there trying to do essentially the same as this one. Ask yourself if you want people to use this project, or the others.

**Reply by HarelM** [reply]
This discussion got too toxic. Locking it...
